src/core/connection.c

```diff
@@ -5835,10 +5830,10 @@ QuicConnProcessRouteCompletion(
                 QuicTraceLogConnInfo(
                     FailedRouteResolution,
                     Connection,
-                    "Processing failed route completion Path[%hhu]",
+                    "Route resolution failed on Path[%hhu]. Switching paths...",
                     PathId);
                 QuicPathSetActive(Connection, &Connection->Paths[1]);
-                QuicSendQueueFlush(&Connection->Send, REASON_ROUTE_COMPLETION);
+                (void)QuicSendFlush(&Connection->Send);
             }
             QuicPathRemove(Connection, PathIndex);
```

Comment:
This seems like another bug. If the above check `Path->IsActive && Connection->PathsCount > 1` passed, the active path we just moved up from index 1 to index 0 would be delete.

I think it should be changed to this:
```
            if (Path->IsActive && Connection->PathsCount > 1) {
                QuicTraceLogConnInfo(
                    FailedRouteResolution,
                    Connection,
                    "Processing failed route completion Path[%hhu]",
                    "Route resolution failed on Path[%hhu]. Switching paths...",
                    PathId);
                QuicPathSetActive(Connection, &Connection->Paths[1]);
                QuicSendQueueFlush(&Connection->Send, REASON_ROUTE_COMPLETION);
                (void)QuicSendFlush(&Connection->Send);
                QuicPathRemove(Connection, 1);
            } else {
                QuicPathRemove(Connection, PathIndex);
           }
```

Reply:
Also, QuicConnProcessRouteCompletion is called with a path ID, which is different from the path index, right? It seems like there's sort of an assumption here that the path QuicConnProcessRouteCompletion is being passed is the path at index 0; is that really a safe assumption?

Reply:
Oh no I see, Path->IsActive==TRUE implies that the path is at index 0.

Reply:
I'm trying to decide which is better: either your suggested fix above, or instead of
                QuicPathSetActive(Connection, &Connection->Paths[1]);
                QuicPathRemove(Connection, 1);

we could do
                QuicPathRemove(Connection, PathIndex);
                QuicPathSetActive(Connection, &Connection->Paths[0]);

Reply:
The former seems slightly more readable to me.

Reply:
Sounds good to me.